apps/encryption/lib/KeyManager.php

```diff
@@ -161,17 +169,24 @@ public function __construct(
 	public function validateShareKey() {
 		$shareKey = $this->getPublicShareKey();
 		if (empty($shareKey)) {
-			$keyPair = $this->crypt->createKeyPair();
-
-			// Save public key
-			$this->keyStorage->setSystemUserKey(
-				$this->publicShareKeyId . '.publicKey', $keyPair['publicKey'],
-				Encryption::ID);
-
-			// Encrypt private key empty passphrase
-			$encryptedKey = $this->crypt->encryptPrivateKey($keyPair['privateKey'], '');
-			$header = $this->crypt->generateHeader();
-			$this->setSystemPrivateKey($this->publicShareKeyId, $header . $encryptedKey);
+			$this->lockingProvider->acquireLock('encryption-generateSharedKey', ILockingProvider::LOCK_EXCLUSIVE, 'Encryption: shared key generation');
+			try {
+				$keyPair = $this->crypt->createKeyPair();
+
+				// Save public key
+				$this->keyStorage->setSystemUserKey(
+					$this->publicShareKeyId . '.publicKey', $keyPair['publicKey'],
```

Comment:
```suggestion
					$this->publicShareKeyId . '.' . $this->publicKeyId, $keyPair['publicKey'],
```

Something like this? Or am I wrong here?

Reply:
? I don't get what you want to suggest.

Reply:
Ah I see. Mainly replace the hardcoded string with the property that contains it ;)

Reply:
> Ah I see. Mainly replace the hardcoded string with the property that contains it ;)

Yes - because you did this at most of the other places as well.